iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1011,10 +1011,13 @@ public void alterTableSnapshotRefOperation(org.apache.hadoop.hive.ql.metadata.Ta
         IcebergSnapshotRefExec.createBranch(icebergTable, createBranchSpec);
         break;
       case CREATE_TAG:
-        Optional.ofNullable(icebergTable.currentSnapshot()).orElseThrow(() -> new UnsupportedOperationException(
-            String.format("Cannot alter %s on iceberg table %s.%s which has no snapshot",
-                alterTableSnapshotRefSpec.getOperationType().getName(), hmsTable.getDbName(),
-                hmsTable.getTableName())));
+        // https://errorprone.info/bugpattern/ReturnValueIgnored
+        if (!Optional.ofNullable(icebergTable.currentSnapshot()).isPresent()) {
```

Comment:
i think it's much more straightforward just to compare with NULL

Reply:
@Aggarwal-Raghav can you please check this comment, its is part of [errorProne upgrade](https://github.com/apache/hive/pull/5404/commits/3e031c3866f8fc751fb745f1b348e00368849fc5)

Reply:
Thanks @Aggarwal-Raghav for addressing this, I have updated this as part of https://github.com/apache/hive/pull/5404/commits/2ceea5d485e42290fea0acd8f4b8cf230d307c84